elektronLogging/logger.go

```diff
@@ -0,0 +1,49 @@
+package elektronLogging
+
+import (
+	log "github.com/sirupsen/logrus"
+	. "github.com/spdfg/elektron/elektronLogging/types"
+	"strings"
+	"time"
+)
+
+var config LoggerConfig
+var logger *log.Logger
+var formatter ElektronFormatter
+var ElektronLog *LoggerImpl
+
+func BuildLogger(prefix string) {
+
+	// read configuration from yaml
+	config.GetConfig()
+
+	// create the log directory
+	startTime := time.Now()
+	formatter.TimestampFormat = "2006-01-02 15:04:05"
+	formattedStartTime := startTime.Format("20060102150405")
+	GetLogDir(startTime, "_")
+
+	prefix = strings.Join([]string{prefix, formattedStartTime}, "_")
+
+	//create a single logrus instance and set its formatter to ElektronFormatter
+	logger = log.New()
+	logger.SetFormatter(&formatter)
+
+	// create a chain of loggers
+	head := &LoggerImpl{}
```

Comment:
What is the rationale behind having a dummy node as the head of the chain?

Reply:
Nothing specific. I had kept `ElektronLog` as `LoggerImpl` type. But I have changed to `ConsoleLogger` rather. Works fine now